src/sagemaker/local/image.py

```
@@ -98,6 +110,68 @@ def __init__(self, instance_type, instance_count, image, sagemaker_session=None)
+        """
+
+        self.container_root = self._create_tmp_folder()
+
+        # A shared directory for all the containers;
+        # it is only mounted if the processing script is Local.
+        shared_dir = os.path.join(self.container_root, "shared")
+        os.mkdir(shared_dir)
+
+        data_dir = self._create_tmp_folder()
+        volumes = self._prepare_processing_volumes(
+            data_dir, processing_inputs, processing_output_config
+        )
+
+        # Create the configuration files for each container that we will create.
+        for host in self.hosts:
+            _create_processing_config_file_directories(self.container_root, host)
+            self.write_processing_config_files(
+                host, environment, processing_inputs, processing_output_config, processing_job_name
+            )
+
+        self._generate_compose_file(
+            "process", additional_volumes=volumes, additional_env_vars=environment
+        )
+        compose_command = self._compose()
+
+        if _ecr_login_if_needed(self.sagemaker_session.boto_session, self.image):
+            _pull_image(self.image)
+
+        process = subprocess.Popen(
+            compose_command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+        )
+
+        try:
+            _stream_output(process)
+        except RuntimeError as e:
+            # _stream_output() doesn't have the command line. We will handle the exception
+            # which contains the exit code and append the command line to it.
+            msg = "Failed to run: %s, %s" % (compose_command, str(e))
+            raise RuntimeError(msg)
+        finally:
+            # Uploading processing outputs back to Amazon S3.
+            self._upload_processing_outputs(data_dir, processing_output_config)
+
+            # Deleting temporary directories.
+            dirs_to_delete = [shared_dir, data_dir]
+            self._cleanup(dirs_to_delete)
```

these cleanups in the finally could fail. do you want to wrap in a `try..except: pass`?

---

Done.